### PR TITLE
Refactor start-training.py into functions

### PR DESCRIPTION
**Pull request by pde**
Refactor `start-training.py`

Everything is now in manageably-sized functions, and the way they affect state during launch is much cleaner to observe.

### CODE REVIEW

File: start-training.py [hunk 1]
```diff
@@ -18,252 +18,296 @@
...
+            if response == 'overwrite':
+                print("Overwriting old data.")
+                shutil.rmtree(li.data_dir)
+            elif response == 'abort':
+                print("Aborting.")
+                exit()
+    else:
+        li.data_dir = li.job_name = None
+
+    setup_config_and_wandb(args, li)
```
Comment (clwainwright) [inline start-training.py:193]: One of the things I liked about doing it the previous way, without functions, was that all of the code in the start-training script was executed strictly in order. Now, when reading the code, you kind of have to read it out of order. For instance, when reading `setup_config_and_wandb` by itself, it's not obvious what the `launch_ingredients` object is and you basically need to skip down to this function to find out.

**Review by clwainwright — COMMENTED** [review]
I don't know, it kind of feels like the `start-training` script started off seeming complicated because it was trying to do a bunch of things, and it's still trying to do the same bunch of things so it still seems complicated, just now with wrapper functions. Global state is slightly more manageable -- there are fewer variables in the global namespace, but now a bunch of things that were in the global namespace are just in a nested namespace, which is tantamount to the same thing. The tradeoff is that we've now got out-of-order execution and shared state between functions. Not sure it's worth it.

Let's chat more about it tomorrow.

**Comment by clwainwright** [comment]
@pde I've made a bunch of changes to break it down into smaller functions with the goal of reducing the amount of state that needs to be passed between functions. I think it's a fair bit more readable now, although the main function is slightly more complicated. I was able to do away with `launch_ingredients` entirely while keeping the call signatures pretty compact. As a side-benefit, the most natural way to do this was to make call tree only one-level deep, so the execution order is essentially as it's written. Let me know what you think. Happy to merge at this point.

**Comment by pde** [comment]
I like most of these changes! Probably the only thing I'd have done differently is to keep the import block at the top, and build the modules inline in the middle of the import.

Merging!

**Pull request merged**